tests/benchq/resource_estimation/test_graph_compilation.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+import pytest
+from orquestra.quantum.circuits import CNOT, RX, RY, RZ, Circuit, H, T
+
+from benchq.data_structures.hardware_architecture_models import BasicArchitectureModel
+from benchq.data_structures.quantum_program import QuantumProgram
+from benchq.resource_estimation.graph_compilation import (
+    get_resource_estimations_for_program,
+)
+
+
+@pytest.mark.parametrize(
+    "quantum_program,expected_results",
+    [
+        (
+            QuantumProgram(
+                [Circuit([H(0), RZ(np.pi / 4)(0), CNOT(0, 1)])], 1, lambda x: [0]
+            ),
+            {"n_measurement_steps": 3, "n_nodes": 3, "max_graph_degree": 2},
+        ),
+        (
+            QuantumProgram(
+                [Circuit([RX(np.pi / 4)(0), RY(np.pi / 4)(0), CNOT(0, 1)])],
+                1,
+                lambda _: [0],
+            ),
+            {"n_measurement_steps": 3, "n_nodes": 4, "max_graph_degree": 2},
+        ),
+        (
+            QuantumProgram(
+                [Circuit([H(0)] + [CNOT(i, i + 1) for i in range(3)])],
+                1,
+                lambda _: [0],
+            ),
+            {"n_measurement_steps": 4, "n_nodes": 4, "max_graph_degree": 3},
+        ),
+        (
+            QuantumProgram(
+                [Circuit([H(0)] + [CNOT(i, i + 1) for i in range(3)] + [T(1), T(2)])],
+                1,
+                lambda _: [0],
+            ),
+            {"n_measurement_steps": 6, "n_nodes": 6, "max_graph_degree": 5},
```

Comment:
I think in this case should have max graph degree equal to 3? If this statement is correct, it should  also has lower `n_measurement_steps` ?

Reply:
I just talked with @AthenaCaesura and it looks ok (90% confidence), but she would need to do some math to be 100%. And also perhaps come with an extra test case.

Reply:
yup! it looksl like you've got the correct max graph degree here! Feel free to merge!